components/gripper/trossen/trossen.go

```diff
@@ -3,58 +3,53 @@ package trossen
 
 import (
 	"context"
-	"sync"
-	"time"
+	"errors"
 
 	"github.com/edaniels/golog"
-	"github.com/jacobsa/go-serial/serial"
-	"go.uber.org/multierr"
-	"go.viam.com/dynamixel/network"
-	"go.viam.com/dynamixel/servo"
-	"go.viam.com/dynamixel/servo/s_model"
-	"go.viam.com/utils"
 
+	"go.viam.com/rdk/components/arm"
+	trossenarm "go.viam.com/rdk/components/arm/trossen"
 	"go.viam.com/rdk/components/generic"
 	"go.viam.com/rdk/components/gripper"
 	"go.viam.com/rdk/config"
-	"go.viam.com/rdk/operation"
 	"go.viam.com/rdk/referenceframe"
 	"go.viam.com/rdk/registry"
 	"go.viam.com/rdk/resource"
 	rdkutils "go.viam.com/rdk/utils"
 )
 
+// This is an implementation of the Gripper API for Trossen arms. The gripper
+// requires a single Trossen arm component in its dependencies upon configuration,
+// as all of the gripper functions defer to DoCommand on the arm implementation. See
+// components/arm/trossen/trossen.go for more about the arm implementations
+// and supported arms
+
 var (
 	modelNameWX250s = resource.NewDefaultModel("trossen-wx250s")
 	modelNameVX300s = resource.NewDefaultModel("trossen-vx300s")
 )
 
-// AttrConfig is the config for a trossen gripper.
+// AttrConfig is the config for a trossen gripper. WARNING: These
+// attributes no longer do anything and should be removed in a
+// later commit - GV.
 type AttrConfig struct {
-	SerialPath string `json:"serial_path"`
-	BaudRate   int    `json:"serial_baud_rate"`
+	SerialPath string `json:"serial_path,omitempty"`
+	BaudRate   int    `json:"serial_baud_rate,omitempty"`
 }
 
 // Validate ensures all parts of the config are valid.
 func (config *AttrConfig) Validate(path string) error {
-	if config.SerialPath == "" {
-		return utils.NewConfigValidationFieldRequiredError(path, "serial_path")
-	}
-
-	if config.BaudRate == 0 {
-		return utils.NewConfigValidationFieldRequiredError(path, "baud_rate")
-	}
 	return nil
```

Comment:
[major] we now need to use a `Validate(path string) ([]string, error)` function to check the dependencies for the gripper. See:
https://github.com/viamrobotics/rdk/blob/e95a3d93252cb4e16ffbcfe1453d6583a14dcffb/components/base/wheeled/wheeled_base.go#L38

Reply:
we're doing this check in initialization, are you saying the code should move to Validate? (I'm comfortable doing that, just clarifying)

Reply:
I think it's better to always check in validate and catch things in the configuration step before anything actually starts doing anything. I'm willing to let it go if you feel like the hardware driver is in a good enough state.

Reply:
Validation can throw errors before the robot even tries to start the part, so any bad user input (e.g. forgetting a required field, etc.) should live there. I believe that App uses this as well, so can check configs as they're being created even.

But yeah, agreeing with Rand... won't block on this at all.

Reply:
I agree in principle, will move the logic to Validate